EIPS/EnsIP01.md

```diff
@@ -0,0 +1,63 @@
+
+**Side note** *Simplicity is not necessarily the opposite of complexity*, specially when we are talking about different levels. Sometimes the most apparently simple looking systems have a great deal of complexity underneath, invisible to most users.
+
+### Why keep Vickrey Auctions at all?
+
+One way to deal with the situation would be to remove Vickrey Auctions alltogether. For example the [Domain Sale contract](https://github.com/wealdtech/domainsale/blob/master/contracts/DomainSale.sol) created by Jim McDonald (with some inputs by myself) uses a standard auctions system that expires after 24h after the last bid was made. This means that if a domain goes undisputed, the buyer needs only to make a single transaction and wait. Standard auctions are known to create inflated prices, specially if bidders get into emotional bidding wars, which might be good for the seller, but since there isn't a concept of original seller in ENS, this isn't necessarily a good thing. Ideally we want users to pay rational utility price for domains.
+
+Also, not necessarily the initial allocations needs to be the most common way to acquire a domain: if a system like Jim's becomes very popular for secondary markets, then we might be able to see a competition betweeen multiple models, which is good in the long run.
+
+## Overview
+
+The purpose of this document is to explore a way in which the properties of a blind vickrey auction can be kept, while removing the requirement of the user to reveal it a posteriori, by making bids as time-locked messages that can be "mined" by a network of computers. The process would be as follows:
+
+1) **Encoding the bid** The potential bidder decides a name and then creates a random private key, with a small amount of bytes according to a **difficulty factor** given by the contract. Then it uses this small key to encrypt a message containing the name it wants and the amount it is willing to pay for it.
+
+2) **Bidding** the bidder then makes a transaction to the chain with an amount of ether equal to or larger than their secret price. The contract doesn't need to store the full message, only a hash of it, the rest of the message can either be given off-chain to "miners" or it can be logged into an event on-chain to ensure it's available for everyone. The bidder also must reveal the size of the key they picked.
+
+3) **Unlocking** anyone is able to "reveal" the bid by revealing the private key. It can be done by the bidder themselves or by other machines by bruteforcing them. Since the owner has revealed the size of the key, machines can decide not to try to unlock a message if they believe it will not be able to do so
+
+4) **difficulty adjustments** If a key is found in less than 48 hours, the **difficulty** factor is slightly increased, in order to target a median of 48 hours. And in the same manner, whenever a bid is revealed after 48h the difficulty should also decrease ever so slightly
```

Comment:
How do you tell the difference between keys found by brute force effort and keys revealed by their creators to game the difficulty?

Reply:
Only bids with a length smaller than the difficulty would be valid for adjustments. Someone holding a private key and revealing it at a specific time can only increase the difficulty, never decrease it, because the keys can be found before by someone else. I don't see how increasing the difficulty can be good for anyone but it's indeed an attack vector